### ISSUE TITLE
[🐛 Bug]: Web worker failing tests

### ISSUE THREAD

**Issue by admc** [webdriverio/webdriverio#11575]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

^8.16.4

### Node.js Version

v18.17.1

### Mode

Standalone Mode

### What happened?

Adding the usage of a web worker seems to break the test for a page that otherwise renders properly, here's the error:

[0-0] Unhandled error caught: Uncaught ReferenceError: wdioImport is not defined Uncaught ReferenceError: wdioImport is not defined
[0-0] Global JS error caught: {
  message: 'Uncaught ReferenceError: wdioImport is not defined',
  source: 'http://localhost:54060/src/utils/ansi.ts',
  lineno: 2,
  colno: 18,
  error: null
}

Here's a code snippet:

`  if (typeof Worker !== 'undefined') {
            const worker = new Worker(
                new URL('./ansiWorker.ts', import.meta.url),
                {
                    type: 'module',
                }
            )
            worker.onmessage = (event) => {
                setOutputHtml(event.data)
            }`

### What is your expected behavior?

Was hoping it would render the way that it does when just using hooks on and a function call.

### How to reproduce the bug.

- vite react typescript webapp
- use a web worker to set a state variable in a component.
- load the page with a test

### Relevant log output

```typescript
2023-11-02T04:57:05.776Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,./wdio.conf.mts,--spec,./src/tests/index.test.tsx
2023-11-02T04:57:09.311Z INFO @wdio/browser-runner:plugin: Received request for: /?cid=0-0&spec=/Users/adam/Projects/stateful/cloud/src/tests/index.test.tsx
2023-11-02T04:57:12.428Z INFO @wdio/browser-runner:ViteServer: Return command result: {"type":2,"value":{"id":"1","result":{"height":1080,"width":1920,"x":0,"y":40}}}
2023-11-02T04:57:47.174Z INFO @wdio/browser-runner:plugin: Received request for: /error?message=Uncaught%20ReferenceError%3A%20wdioImport%20is%20not%20defined
2023-11-02T04:57:47.224Z INFO @wdio/browser-runner:plugin: Received request for: /error?message=Uncaught%20ReferenceError%3A%20wdioImport%20is%20not%20defined
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!

Interesting. We don't yet instrument the environment for worker threads. I am sure it is possible to detect a worker thread request and inject the necessary instrumentation into it.

We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by admc** [comment]
Makes sense, have been working on mocking it out, but can't quite get it. It would be a great example for the mock docs!

**Comment by christian-bromann** [comment]
The issue is that we register our import handler (`wdioImport`) directly in the [test file](https://github.com/webdriverio/webdriverio/blob/ecd74b42b0fa67be1d9fcd83ba9be616ed266621/packages/wdio-browser-runner/src/vite/utils.ts#L68-L92) instead of in each file. Furthermore we also register mocks within the test file which will be hard to propagate that into a web worker environment. For now I'ld suggest to ignore the [import transformations](https://github.com/webdriverio/webdriverio/blob/ecd74b42b0fa67be1d9fcd83ba9be616ed266621/packages/wdio-browser-runner/src/vite/plugins/mockHoisting.ts#L169-L293) in web worker files to fix this issue and come up with a solution for dependency mocking in web workers later.

**Comment by admc** [comment]
Ah okay interesting, that link seems to 404, is there somewhere else I
could be looking?

**Comment by christian-bromann** [comment]
updated links
